### ISSUE TITLE
The no-dupe-keys rule complains about legal duplicated keys in a declared class

### ISSUE THREAD

**Issue by object88** [gajus/eslint-plugin-flowtype#133]
Take, for instance, this Flow class:

``` Javascript
//@flow
...
declare class ProjectClass extends Map<any, any> {
  get(key: 'id'): number;
  get(key: 'children'): List<Project>;
}
```

The ESLint engine complains, 

```
flowtype/no-dupe-keys Duplicate property.: /Users/pairing/[...]/ProjectModalBody.js:1
```

However, it's allowed to have two `get` methods, as they are differentiated with different `keys`.  It would be great for the dupe rule to be aware of this and allow it.

**Comment by danharper** [comment]
👍 good spot

**Comment by object88** [comment]
I am starting to take a look at a fix now.  A little more complicated than I had initially expected, but will comment here if I get stuck or have something to merge.
